docs/netserver/config/clientconfigurationprovider.md

```diff
@@ -1,29 +1,91 @@
-description: NetServer ClientConfigurationProvider element
-so.date: 06.06.2018
-author: {github-id}
-keywords: config
+description: NetServer configuration for the Web client configuration data provider.
+so.date: 12.08.2021
+author: Bergfrid Dias
+keywords: config, NetServer, web.config, ClientConfigurationProvider, FilePath, CacheConfigurations, CacheUserPreferences, ValidateConfigurations, CustomPath
 so.topic: reference
 so.envir: onsite
 so.client: web
 ---
 
 # NetServer ClientConfigurationProvider element (SuperOffice only)
 
-Configuration for the Web Client configuration data provider.
+Configuration for the Web client configuration data provider. The paths tell the client where additional configuration files are located.
 
 ```XML
 <ClientConfigurationProvider>
   <add key="CacheConfigurations" value="false" />
+  <add key="ValidateConfigurations" value="true" />
 </ClientConfigurationProvider>
 ```
 
-| Name | Description |
-|---|---|
-| FilePath | File path for standard .config files. |
-| CacheConfigurations | Should GUI config data be cached? |
-| CacheUserPreferences | Should GUI config data be cached?<br>Default: true |
-| ValidateConfigurations | Should GUI config data be validated runtime? |
-| CustomPath | List of alternate paths for GUI .config files. When more than one, each additional path must have a numbered suffix (CustomPath1, CustomPath2, ...) |
+| Name | Description | Default |
+|---|---|---|
+| CacheConfigurations | Should GUI config data be cached? | |
+| CacheUserPreferences | Should user preferences be cached? | true |
+| CustomPath | List of alternate paths for GUI .config files. When more than one, each additional path must have a numbered suffix (CustomPath1, CustomPath2, ...) | |
+| FilePath | File path for standard .config files. | |
+| ValidateConfigurations | Should GUI config data be validated runtime? | |
+
+## CacheConfigurations
+
+Caching configuration benefits performance. However, it makes more sense to turn caching off during development.
+
+When set to false, configuration changes to UI elements are immediately applied and observed in the browser. There's no need to reset IIS or issue a *flush* SoProtocol.
+
+> [!NOTE]
+> Caching works for all configuration files **except system files**. These are cached during application start-up and held in memory.
+
+Any changes to the following files do require an IIS reset:
+
+* SoApplicationConfiguration.config
+* SoAdminApplicationConfiguration.config
+* SoFilterList.config
+* SoObjectMapping.config
+* SoArchiveColumnList.config
+* SoArchiveControlLinkInfoTypes.config
+* SoArchiveCriteriaList.config
+
+> [!NOTE]
+> Any changes to the `archive` configuration will not be observed in the client until the corresponding records in the database are purged.
+
+Once an archive provider is initialized with column definitions in an `archive` element, the column definitions are persisted like preferences in the `SUPERLISTCOLUMNSIZE` table of the database.
```

Comment:
This information pertains to the creation/building of a custom archive provider and control. It makes more sense to have this note (make it a warning) in the [Create a custom archive control](http://localhost:8080/ui/web-app/tutorials/custom-archive-control/index.html) text.